lib/settings_base.py

```diff
@@ -80,6 +80,9 @@ def path(*folders):
 DATABASES['default']['TEST_CHARSET'] = 'utf8'
 DATABASES['default']['TEST_COLLATION'] = 'utf8_general_ci'
 
+# Run all views in a transaction unless they are decorated not to.
+ATOMIC_REQUESTS = True
```

Comment:
This might not matter but I don't understand how it would work so I'm commenting.

The way I understand it `ATOMIC_REQUESTS` is essentially going to turn off auto-commit. How does django know to select the right database in our primary/replica setup though?

I would imagine that if the first query is a select it will happen on the replica and then once you write you get moved over to the primary database for the remainder of the request (and maybe some subsequent requests).

Do we need to tell django to start these atomic requests using the primary database? That seems like it would dramatically increase the load on that database but I don't understand how transactions could work across the two databases.